### PR TITLE
FM2-617: Use system to differentiate patient identifer types

### PR DESCRIPTION
**Pull request by k4pran**
## Description of what I changed

Enables retrieval of PatientIdentifierType objects by their URL via the system, which takes precedence over the identifier type text when available.

## Issue I worked on

see https://issues.openmrs.org/browse/FM2-617

## Checklist: I completed these to help reviewers :)

- [X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [X] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [X] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [X] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/module/fhir2/api/impl/FhirPatientServiceImplTest.java [hunk 1]
```diff
@@ -873,6 +887,68 @@ public void getPatientEverything_shouldReturnAllInformationAboutAllPatients() {
...
+		PatientIdentifierType result = patientService.getPatientIdentifierTypeByIdentifier(identifier);
+		assertNull(identifier.getSystem());
+		assertFalse(identifier.hasType());
+		assertNull(result);
+	}
+	
+	@Test
+	public void getPatientIdentifierTypeByIdentifier_shouldReturnNullWhenIdentifierTypeTextEmpty() {
+		Identifier identifier = new Identifier();
+		identifier.setType(new CodeableConcept().setText(Strings.EMPTY));
```
Comment (mherman22) [inline api/src/test/java/org/openmrs/module/fhir2/api/impl/FhirPatientServiceImplTest.java:944]:
```suggestion
		identifier.setType(new CodeableConcept().setText(null));
```
any reason for not using null instead (as shown in the suggestion above)?

**Review by mherman22 — COMMENTED** [review]
Thanks @k4pran for your work. just a question below

File: api/src/test/java/org/openmrs/module/fhir2/api/impl/FhirPatientServiceImplTest.java (hunk 1 above)
Comment (k4pran) [reply api/src/test/java/org/openmrs/module/fhir2/api/impl/FhirPatientServiceImplTest.java:966]: Thanks @mherman22 for your suggestion. No reason for it, I was just trying to test the statement:

`StringUtils.isBlank(identifier.getType().getText()`

You have a good point that a null test would be good to check it is handled correctly too, so I have added a separate test for that now. I also realised there is no test for the case where the Identifier.type is null so also added a test case for this.

**Review by ibacher — APPROVED** [review]
Awesome! Thanks @k4pran!

**Pull request merged**